mysql/tests/conftest.py

```diff
@@ -57,16 +57,37 @@ def dd_environment(config_e2e):
                 'MYSQL_LOGS_PATH': logs_path,
                 'WAIT_FOR_IT_SCRIPT_PATH': _wait_for_it_script(),
             },
-            conditions=[
-                WaitFor(init_master, wait=2),
-                WaitFor(init_slave, wait=2),
-                CheckDockerLogs('mysql-slave', ["ready for connections", "mariadb successfully initialized"]),
-                populate_database,
-            ],
+            conditions=build_wait_conditions(),
         ):
             yield config_e2e, e2e_metadata
 
 
+def build_wait_conditions():
+    master_logs = []
+    # The order matters for log conditions
+    if MYSQL_FLAVOR == 'mariadb':
+        master_logs.append("MariaDB setup finished!")
+        master_logs.append("Starting MariaDB")
+    else:
+        if common.MYSQL_VERSION_PARSED < parse_version('8.0'):
+            master_logs.append("MySQL init process done")
+            master_logs.append("ready for connections")
+        else:
+            master_logs.append("MySQL setup finished!")
+            master_logs.append("Starting MySQL")
+
+    conditions = []
+    for log in master_logs:
+        conditions.append(CheckDockerLogs(common.MASTER_CONTAINER_NAME, [log]))
+    conditions.append(
+        CheckDockerLogs(common.SLAVE_CONTAINER_NAME, ["ready for connections", "mariadb successfully initialized"])
+    )
+    conditions.append(WaitFor(init_master, wait=2))
+    conditions.append(WaitFor(init_slave, wait=2, attempts=180))
```

Comment:
Note: that's a bit long, but the slave might takes a long time to startup on CI.